migration/export_database.py

```diff
@@ -0,0 +1,142 @@
+        Returns 1 if some other error occurs
+    '''
+    try:
+        mc_new.get_default_database().notices.insert(notice)
+        print("inserted specific notice: ", notice)
+        return 0
+    except errors.DuplicateKeyError:
+        print("specific entry already in database")
+        return 2
+    except errors.ConnectionFailure:
+        print("Error while connecting to DB")
+        return 1
+    except Exception as ex:
+        print("error in specific inserting: ", ex)
+        return 1
+
+def insert_from_file(filename):
+    '''
+        Reads a bson file with name 'filename' and tries to insert each notice in the bson to the target Database
+
+        If a notice fails to be inserted it is pushed to further_defaulters and finally saved in file with filename 'further_defaulter_filename'
+        If a notice already exists in target database, it is pushed to 'further_repeated_filename'
+    '''
+    further_defaulter_filename = "further_defaulters.bson"
+    further_repeated_filename = "further_repeated.bson"
+
+    mc_new = MongoClient(new_mongodb_uri)
+    print("connected to new DB")
+
+    further_defaulters = []
+    further_repeated = []
+    with open(filename, "r") as f:
+        notices = loads(f.read())
+        print("Notice count: {}".format(len(notices)))
+        for notice in notices:
+            out = insert_notice(notice, mc_new)
+            if(out == 1):
+                further_defaulters.append(notice)
+            elif(out == 2):
+                further_repeated.append(notice)
+
+    print("Further defaulters count: {}".format(len(further_defaulters)))
+    print("Further repeated count: {}".format(len(further_repeated)))
+
+    print("Saving further defaulters to {}".format(further_defaulter_filename))
+    with open(further_defaulter_filename, "w") as f:
+        f.write(dumps(further_defaulters))
+
+    print("Saving further repeated to {}".format(further_repeated_filename))
+    with open(further_repeated_filename, "w") as f:
+        f.write(dumps(further_repeated))
+
+    print("Attempt to insert from file: {} complete".format(filename))
+
+def start_database_export():
+    '''
+        Script for exporting database
+        Using OLD_MONGODB_URI in env to act as original database
+        Add NEW_MONGODB_URI in env to act as target database
+    '''
+    export_db()
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Database Export Script")
+    parser.add_argument('--target', '-t', dest="new_mongodb_uri", help="URI for target database", required=True)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('--source', '-s', dest="old_mongodb_uri", help="URI for source database")
+    group.add_argument('--try-defaulters', '-d', dest="defaulter_filename", help="bson file with defaulters dump to upload")
+
+    args = parser.parse_args()
+
+    old_mongodb_uri = args.old_mongodb_uri
+    new_mongodb_uri = args.new_mongodb_uri
```

Comment:
Will this not give error if `--source` is not set in case of using `--try-defaulters`?

Reply:
No, in case the old_mongodb_uri is not supplied, this will be None or something. It'll not throw error, I checked.

Reply:
Cool, thanks for helping me understnad.